docs/aws/README.md

```diff
@@ -37,54 +38,47 @@ SUBNET_IDS=$(aws cloudformation describe-stacks \
 
-Note: For IRSA to work your [cluster needs an OIDC provider](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+> For IRSA to work your cluster needs an [OIDC provider](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
 
 ```bash
-OIDC_PROVIDER=$(aws eks describe-cluster \
-    --name ${CLUSTER_NAME} \
-    --query 'cluster.identity.oidc.issuer' \
-    --output text)
-
 # Creates IAM resources used by Karpenter
-LATEST_KARPENTER_VERSION=$(curl \
-  https://api.github.com/repos/awslabs/karpenter/releases/latest | jq -r '.tag_name')
 TEMPOUT=$(mktemp)
-curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/"${LATEST_KARPENTER_VERSION}"/docs/aws/karpenter.cloudformation.yaml > $TEMPOUT \
+curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/"${KARPENTER_VERSION}"/docs/aws/karpenter.cloudformation.yaml > $TEMPOUT \
 && aws cloudformation deploy \
   --stack-name Karpenter-${CLUSTER_NAME} \
   --template-file ${TEMPOUT} \
   --capabilities CAPABILITY_NAMED_IAM \
-  --parameter-overrides ClusterName=${CLUSTER_NAME} OpenIDConnectIdentityProvider=${OIDC_PROVIDER/https:\/\//}
-
-# Adds the karpenter node role to your aws-auth configmap, allowing nodes with this role to connect to the cluster.
-kubectl patch configmap aws-auth -n kube-system --patch "$(cat <<-EOM
-data:
-  mapRoles: |
-    - rolearn: arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterNodeRole-${CLUSTER_NAME}
-      username: system:node:{{EC2PrivateDNSName}}
-      groups:
-      - system:bootstrappers
-      - system:nodes
-$(kubectl get configmap -n kube-system aws-auth -ojsonpath='{.data.mapRoles}' | sed 's/^/    /')
-EOM
-)"
+  --parameter-overrides ClusterName=${CLUSTER_NAME}
+
+# Add the karpenter node role to your aws-auth configmap, allowing nodes with this role to connect to the cluster.
+eksctl create iamidentitymapping \
+  --username system:node:{{EC2PrivateDNSName}} \
+  --cluster  ${CLUSTER_NAME} \
+  --arn arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterNodeRole-${CLUSTER_NAME} \
+  --group system:bootstrappers \
+  --group system:nodes
 ```
 
 ### Install Karpenter
+
+Use [`helm`](https://helm.sh/) to deploy Karpenter to the cluster.
+For additional values, see [the helm chart values](https://github.com/awslabs/karpenter/blob/main/charts/karpenter/values.yaml)
+
+> We created a Kubernetes service account with our cluster so we don't need the helm chart to do that.
+
 ```bash
 helm repo add karpenter https://awslabs.github.io/karpenter/charts
 helm repo update
-# For additional values, see https://github.com/awslabs/karpenter/blob/main/charts/karpenter/values.yaml
-helm upgrade --install karpenter karpenter/karpenter --create-namespace --namespace karpenter \
-  --set serviceAccount.annotations.'eks\.amazonaws\.com/role-arn'=arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterControllerRole-${CLUSTER_NAME}
```

Comment:
IIUC we need to annotate this here so that Karpenter has the appropriate permissions to take action. Am I missing something here?

Reply:
by using eksctl config to create the service account it automatically gets the ARN annotation needed for the IAM role. The chart uses a default serviceaccount named `karpenter` so we created one in the eks-config.yaml file. It also handles the trust for OIDC for IAM role so there was less for us to manually configure.